### DISCUSSION TITLE
How to Create Server and API key in app.shlink.io?

### DISCUSSION THREAD

**Discussion by dt2036** [shlinkio/discussions#38] [General]
How to generate a Server? we face the below error: Oops! Could not connect to this Shlink server.
Make sure you have an internet connection, and the server is properly configured and online.
![image](https://user-images.githubusercontent.com/69676290/118134497-9fa04a00-b41f-11eb-8c09-12609de78d0b.png)
and when we use the existing API key then give an invalid API key error.
[](url)
![MicrosoftTeams-image (14)](https://user-images.githubusercontent.com/69676290/118135076-37059d00-b420-11eb-8425-f47b8255f256.png)

Thanks in advance.

**Comment by acelaya** [comment]
You need to install and host the server yourself:

* Via docker image: https://shlink.io/documentation/install-docker-image/
* Via dist file: https://shlink.io/documentation/install-dist-file/

**Reply by dt2036** [reply]
Thanks for the quick reply. @acelaya 

when we install and host the server vis docker image using docker **run --name my_shlink -p 8080:8080 -e SHORT_DOMAIN_HOST=doma.in -e SHORT_DOMAIN_SCHEMA=https -e GEOLITE_LICENSE_KEY=nCsg6IAezeH6CkDs shlinkio/shlink:stable** 

stuck here, take 2 hours but installation not complete.

![image](https://user-images.githubusercontent.com/69676290/118227817-ce153800-b4a6-11eb-8b15-fae6b091350f.png)
Thanks in advance.

**Reply by acelaya** [reply]
What do you mean by "stuck"?

That's a perfectly working instance serving requests.

**Reply by dt2036** [reply]
frozen or fixed in one place and can't be moved.

How much time to take complete the installation?
**run --name my_shlink -p 8080:8080 -e SHORT_DOMAIN_HOST=doma.in -e SHORT_DOMAIN_SCHEMA=https -e GEOLITE_LICENSE_KEY=nCsg6IAezeH6CkDs shlinkio/shlink:stable** 

After completing the above step we can generate API Key using run command  **docker exec -it my_shlink shlink api-key:generate** in cmd.
Right?

**Reply by acelaya** [reply]
Yes, I know what the stuck word means 😄

I mean that what makes you think it is stuck. According to your screenshot, it's working perfectly fine.

**Reply by dt2036** [reply]
How much time to take complete the installation?

**Reply by acelaya** [reply]
As I said, your instance is already working.

**Reply by dt2036** [reply]
How to generate API Key for Rest API and UI app.shlink.io?

**Reply by dt2036** [reply]
Got API Key 
![image](https://user-images.githubusercontent.com/69676290/118236256-7af5b200-b4b3-11eb-8e74-fb6d3a21526b.png)

Thanks.

**Reply by dt2036** [reply]
I just wanted to know On which basis are you tracking visits? GPS or IP address? Is Shlink providing accurate results if the location is off?
